src/lib.rs

```diff
@@ -53,11 +60,91 @@ pub use hashes::*;
 ///
 pub fn encode(wanttype: HashTypes, input: &[u8]) -> io::Result<Vec<u8>> {
     let digest: Vec<u8> = match wanttype {
-        HashTypes::SHA2256 => sha256::hash(input).as_ref().to_owned(),
-        HashTypes::SHA2512 => sha512::hash(input).as_ref().to_owned(),
+
+        HashTypes::SHA1 => {
+            let mut buf : Vec<u8> = Vec::new();
+            let mut hasher = Sha1::new();
+            hasher.input(input);
+            hasher.result(&mut buf);
+            buf
+        },
+
+        HashTypes::SHA2256  => {
+            let mut buf : Vec<u8> = Vec::new();
+            let mut hasher = Sha256::new();
+            hasher.input(input);
+            hasher.result(&mut buf);
+            buf
+        },
+
+        HashTypes::SHA2512  => {
+            let mut buf : Vec<u8> = Vec::new();
+            let mut hasher = Sha512::new();
+            hasher.input(input);
+            hasher.result(&mut buf);
+            buf
+        },
+
+        HashTypes::SHA3512  => {
+            let mut buf : Vec<u8> = Vec::new();
+            let mut hasher = Sha3::sha3_512();
+            hasher.input(input);
+            hasher.result(&mut buf);
+            buf
+        },
+
+        HashTypes::SHA3384  => {
+            let mut buf : Vec<u8> = Vec::new();
+            let mut hasher = Sha3::sha3_384();
+            hasher.input(input);
+            hasher.result(&mut buf);
+            buf
+        },
+
+        HashTypes::SHA3224  => {
+            let mut buf : Vec<u8> = Vec::new();
+            let mut hasher = Sha3::sha3_224();
+            hasher.input(input);
+            hasher.result(&mut buf);
+            buf
+        },
+
+        HashTypes::SHAKE128 => {
+            let mut buf : Vec<u8> = Vec::new();
+            let mut hasher = Sha3::shake_128();
+            hasher.input(input);
+            hasher.result(&mut buf);
+            buf
+        },
+
+        HashTypes::SHAKE256 => {
+            let mut buf : Vec<u8> = Vec::new();
+            let mut hasher = Sha3::shake_256();
+            hasher.input(input);
+            hasher.result(&mut buf);
+            buf
+        },
+
+        HashTypes::Blake2b  => {
+            let mut buf : Vec<u8> = Vec::new();
+            let mut hasher = Blake2b::new(64);
+            hasher.input(input);
+            hasher.result(&mut buf);
+            buf
+        },
+
+        HashTypes::Blake2s  => {
+            let mut buf : Vec<u8> = Vec::new();
+            let mut hasher = Blake2s::new(32);
+            hasher.input(input);
+            hasher.result(&mut buf);
+            buf
```

Comment:
you could also preallocate the vec:
```rust
let mut buf : Vec<u8> = Vec::with_capacity(wanttype.size());
```

ps: (IMHO) HashTypes.size should return *usize*